trie/proof.go

```diff
@@ -408,7 +408,16 @@ func VerifyRangeProof(rootHash common.Hash, firstKey []byte, keys [][]byte, valu
 		return fmt.Errorf("inconsistent proof data, keys: %d, values: %d", len(keys), len(values)), false
 	}
 	if len(keys) == 0 {
-		return fmt.Errorf("empty proof"), false
+		return errors.New("empty proof"), false
+	}
+	// Ensure the received batch is monotonic increasing.
+	for i := 0; i < len(keys); i++ {
+		if i+1 == len(keys) {
```

Comment:
Why not `for i := 0; i < len(keys)-1; i++` instead of this extra break?

Reply:
If there is only 1 element, then it's out of bound.

Reply:
Fixed